src/backend/impl/src/services/proposal_service.rs

```diff
@@ -148,6 +179,61 @@ impl<T: ProposalRepository> ProposalServiceImpl<T> {
             proposal_repository,
         }
     }
+
+    async fn fetch_and_complete_missing_proposals(
+        &self,
+        nns_proposals: &[ProposalInfo],
+    ) -> Result<usize, ApiError> {
+        let in_progress_proposals = self
+            .proposal_repository
+            .get_proposals(Some(ReviewPeriodState::InProgress))?;
+        let missing_proposals: Vec<(ProposalId, Proposal)> = in_progress_proposals
+            .iter()
```

Comment:
```suggestion
            .into_iter()
```

Then line 198 can be `Some((internal_id, proposal))`.